daml-lf/kv-support/src/main/scala/com/daml/lf/kv/transactions/TransactionConversions.scala

```diff
@@ -92,9 +96,161 @@ object TransactionConversions {
       .sequence_
       .map(_ => RawTransaction(transactionBuilder.build.toByteString))
   }
+
+  /** Decodes and extracts outputs of a submitted transaction, that is the IDs and keys of contracts created or updated
+    * by processing a submission.
```

Comment:
I think it’s worth documenting that we overapproximate the outputs meaning we include transient contracts, contract ids that only cause divulgence, keys that are not modified, everything under a rollback node, ….

Reply:
Done, I mostly used a comment from below. @cocreature, are you able to help me answer a few questions?
1. Just to make sure, transient contracts are those created and archived within the same transaction, right?
2. Is it correct to say "contract IDs that may cause divulgence", what does it actually mean? Or is it more precise to say "contract IDs of contracts that may be divulged"?
3. Why do we keep "keys that are not modified". Is it just because it's easy or we cannot even say whether they were modified or not?
4. Does "under a rollback node" mean children of a rollback node? I assume we include them because of using all nodes from `transaction.getNodesList`, is that right?

Reply:
1. right
2. we divulge the contract payload not the contract id so arguably the first phrasing is correct.
3. I don’t know, that’s something for your team. But I suspect it’s the same reason why we overapproximate the other stuff: it doesn’t do much harm and changing it adds complexity.
4. Yes, children (or transitive children) of rollback nodes. And your analysis is exactly right.

Reply:
@hubert-da : on "keys that are not modified" -- including these can lead to spurious contention between transactions; and we did observe that for one of our customers which uses keys to maintain some intra-transaction state. I have a vague memory that @miklos-da implemented a post-processing step, which removes writes to keys that are equal to their read.

In general, our code should aim to minimize accidental contention as that negatively impacts our customer's operating experience.

@hubert-da @cocreature : wrt "contract IDs that may cause divulgence" -- these MUST be included in the current state of the world, as the kv-state tracks which contracts have been divulged to whom; and divulgence thus results in write-conflicts. I believe we can change this now, as we use transaction local divulgence (i.e., redivulge all contracts in every transaction independent of the previous state of divulgence). However, it would be better for @miklos-da to coordinate this, as he knows the detailed state of the various changes.

Reply:
divulgence makes sense to me, including transient contracts on the other hand not really.